Updated the OpenStack components install method

costingalan opened the pull request:
This commit adds the requirements check for the components, thus making
sure the requirements are synced by the OpenStack Requirements project
team.

The components affected by this change are:
- nova
- neutron
- compute-hyperv
- networking-hyperv
- os-win

ociuhandu commented on HyperV/scripts/create-environment.ps1:
```diff
@@ -138,8 +138,13 @@ if ($buildFor -eq "openstack/compute-hyperv") {
     }
-    ExecRetry {
-        GitClonePull "$buildDir\os-win" "https://git.openstack.org/openstack/os-win.git" master
+    if ($branchName -eq 'master') {
```
Why this if?

ociuhandu commented on HyperV/scripts/create-environment.ps1:
```diff
@@ -226,15 +231,29 @@ if ($isDebug -eq  'yes') {
+    pushd "$buildDir\os-win"
     if ($branchName.ToLower().CompareTo('master') -eq 0) {
         # only install os-win on master.
-        & pip install $buildDir\os-win
```
and where do u install os-win?

ociuhandu commented on HyperV/scripts/create-environment.ps1:
```diff
@@ -226,15 +231,29 @@ if ($isDebug -eq  'yes') {
     Get-ChildItem $buildDir
 }
 
+if ($branchName -eq 'master')
```
Why only on master?

ociuhandu commented on HyperV/scripts/create-environment.ps1:
```diff
@@ -138,8 +138,13 @@ if ($buildFor -eq "openstack/compute-hyperv") {
+        ExecRetry {
+            GitClonePull "$buildDir\requirements" "https://git.openstack.org/openstack/requirements.git" $branchName
+        }
+        ExecRetry {
```
Based on the "if" containing this, what happens for stable branches?

Commit:
Updated the OpenStack components install method

This commit adds the requirements check for the components, thus making
sure the requirements are synced by the OpenStack Requirements project
team.

The components affected by this change are:

 * nova
 * neutron
 * compute-hyperv
 * networking-hyperv
 * os-win

Signed-off-by: Costin Galan <cgalan@cloudbasesolutions.com>
